### DISCUSSION TITLE
Drag & Drop for switching program output

### DISCUSSION THREAD

**Discussion by SquirrelsMcGee** [File-New-Project/EarTrumpet#1146] [Ideas]
Hi,

Currently in order to switch outputs for specific programs you have to:

1) Right click the program in the list
2) Click the switch-output button (double arrows)
3) Select the output

What I would like to see is the ability to drag & drop programs to each output 

This would be really useful for people (Like myself) who often find themselves switching devices / creating ad-hoc combinations

**Comment by jac0-b** [comment]
Hopefully #1122 will get merged soon (discussion #662)

**Reply by riverar** [reply]
There are still some outstanding design questions. Like, how does a user return an app to the state in which it follows the "default" device?

**Reply by jac0-b** [reply]
I think you're referring to ability to be able to access this button:
![image](https://user-images.githubusercontent.com/51512690/167225759-f4d2c158-d56f-4910-a185-8795e9997c36.png)

Would it not be possible keep the current right click menu but also have the ability to drag and drop? (I haven't actually run the PR so I'm not sure how it's currently implemented)

**Reply by SquirrelsMcGee** [reply]
This is a good idea imo - Fits all the requirements

**Reply by Masurov** [reply]
I added drag&drop in #1122 without altering right click menu ability. I thought about Riverar's remark in the PR, I agree that eartrumpet mixer should show the if the app's device is the default or not. That would be a great addition.

In my opinion, a pin button at the app level mixer would provide insight that the current app is forced to a specific device. The most challenging is to not overload the view and keep it neat. I provided a naive design to illustrate this :
![image](https://user-images.githubusercontent.com/16320196/167428992-fa7eb4c7-df4a-4885-958c-0de1436c1cb3.png)

**Reply by jac0-b** [reply]
@Masurov So every time you drag a source to an output by default it will force it to that device and the pin indicates that. It would also be pretty neat if pressing the pin reverts the source to using the default device.

**Reply by Masurov** [reply]
Yeah, that's my suggestion for dealing with displaying/canceling device specific app association. 

Though my PR #1122 only covers drag and drop and not that "pin" suggestion.
